zanata-war/src/main/java/org/zanata/limits/RateLimitManager.java

```diff
@@ -125,7 +125,7 @@ public String getFieldValueAsString(String fieldName) {
                 });
     }
 
-    public RestCallLimiter getLimiter(final String apiKey) {
+    public RestCallLimiter getLimiter(final String key) {
```

Comment:
> not sure what you meant by "immutable" value class for the key

I was thinking of something like this:

```java
    @Getter
    class RateLimiterKey {
    	private final String username;
    	private final String apiKey;
    	private final String ipAddress;
    	// and three different static factory methods...
    }
```
Or perhaps an enum would be better:
```java
    enum RateLimiterKeyType {
    	USERNAME, API_KEY, IP_ADDRESS
    }

    @Getter
    class RateLimiterKey {
    	private final RateLimiterKeyType keyType;
    	private final String username;
    	private final String apiKey;
    	private final String ipAddress;
    	// toString()
    	// and three different static factory methods...
    }
```
However, that's all getting to be a bit much for such a simple thing.  We don't store these values long term, so we can always change the implementation later if need be.

- [ ] Perhaps we should just call `key` something like `token` instead, just to make it clear that it may not be an API key.